portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -119,7 +103,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
    */
     private String getParameter(HttpServletRequest request, String name, String defaultValue) {
         String value = request.getParameter(name);
-        if (value.equals(null)) {
+        if (value == null) {
```

Comment:
In Java, everything is a reference. For example, "value" variable here is a reference to a String object, and Java provides two types of equality comparison:
(1) checking if the reference is null, meaning whether the reference is pointing to any object. Your "value == null" is doing this;
(2) checking if the object referenced by a variable is indeed equal to the content of a given object, where you would use something like "value.equals("")", which says if the object referenced by variable "value" actually equals to the empty String object.

In this case, I think you should use the second one since the caller always passes "" to defaultValue, meaning that it always points to some object (i.e., value is never null here).